doc/jms.adoc

```diff
@@ -0,0 +1,315 @@
+
+The JMS connector is consuming the produced _factory_ to connect to the broker.
+
+==== Selecting ConnectionFactory
+
+In case you have several connection factories, you can use the `@Named` qualified on your factory to specify the name.
+Then, in the channel configuration, configure the name as follows:
+
+[source,properties]
+----
+# Configure the connector globally
+mp.messaging.connector.smallrye-jms.connection-factory-name=my-factory-name
+# Configure a specific incoming channel
+mp.messaging.incoming.my-channel.connection-factory-name=my-factory-name
+# Configure a specific outgoing channel
+mp.messaging.outgoing.my-channel.connection-factory-name=my-factory-name
+----
+
+==== Configure credentials
+
+To configure the _username_ and _password_ to connect to the broker and create a session, use the following properties:
+
+[source,properties]
+----
+# Configure the connector globally
+mp.messaging.connector.smallrye-jms.username=my-username
+mp.messaging.connector.smallrye-jms.password=secret
+# Configure a specific incoming channel
+mp.messaging.incoming.my-channel.username=my-username
+mp.messaging.incoming.my-channel.password=secret
+# Configure a specific outgoing channel
+mp.messaging.outgoing.my-channel.username=my-username
+mp.messaging.outgoing.my-channel.password=secret
+----
+
+==== Session mode
+
+By default, the connector uses the `AUTO_ACKNOWLEDGE` session mode.
+You can configure it using:
+
+[source,properties]
+----
+# Configure the connector globally
+mp.messaging.connector.smallrye-jms.session-mode=AUTO_ACKNOWLEDGE
+# Configure a specific incoming channel
+mp.messaging.incoming.my-channel.session-mode=AUTO_ACKNOWLEDGE
+# Configure a specific outgoing channel
+mp.messaging.outgoing.my-channel.session-mode=AUTO_ACKNOWLEDGE
+----
+
+Accepted values are:
+
+* `AUTO_ACKNOWLEDGE`
+* `SESSION_TRANSACTED`
+* `CLIENT_ACKNOWLEDGE`
+* `DUPS_OK_ACKNOWLEDGE`
+
+
+==== Configure the client id
+When dealing with _topics_, your subscription must need to be identified with a _client-id_.
+Configure the client id as follows:
+
+[source,properties]
+----
+# Configure the connector globally
+mp.messaging.connector.smallrye-jms.client-id=my-id
+# Configure a specific incoming channel
+mp.messaging.incoming.my-channel.client-id=my-id
+# Configure a specific outgoing channel
+mp.messaging.outgoing.my-channel.client-id=my-id
+----
+
+=== JSON-B Requirement
+
+The connector uses JSON-B to serialize and deserialize objects.
+You need to have an implementation of JSON-B available.
+Check http://json-b.net/ for more details.
+
+=== Sending JMS messages
+
+To send a JMS message, you need to configure the connector as follows:
+
+[source, text]
+----
+# Queue:
+mp.messaging.outgoing.my-channel.connector=smallrye-jms
+mp.messaging.outgoing.my-channel.destination=my-queue
+
+# Topic:
+mp.messaging.outgoing.my-channel.connector=smallrye-jms
+mp.messaging.outgoing.my-channel.destination=my-topic
+mp.messaging.outgoing.my-channel.destination-type=topic
```

Comment:
The spec needs to state an `IllegalArgumentException` occurs if the destination-type is neither `queue` nor `topic`.

Reply:
I'm going to add a sentence in the documentation and a test (because we never know...)

Reply:
Done.